qt/scientific_interfaces/Indirect/ISISCalibration.cpp

```diff
@@ -366,21 +396,24 @@ void ISISCalibration::setDefaultInstDetails(QMap<QString, QString> const &instru
   // Set peak and background ranges
   const auto ranges = getRangesFromInstrument();
 
-  QFileInfo fi(m_lastCalPlotFilename);
+  QString filename = m_uiForm.leRunNo->getFirstFilename();
+  if (filename.isEmpty())
+    return;
+  QFileInfo fi(filename);
   QString wsname = fi.baseName();
+  if (!Mantid::API::AnalysisDataService::Instance().doesExist(wsname.toStdString())) {
+    loadFile(filename, wsname, spectraMin, spectraMax);
+  }
+  const auto input =
+      std::dynamic_pointer_cast<MatrixWorkspace>(AnalysisDataService::Instance().retrieve(wsname.toStdString()));
+  const auto &dataX = input->x(0);
 
-  if (Mantid::API::AnalysisDataService::Instance().doesExist(wsname.toStdString())) {
-    const auto input =
-        std::dynamic_pointer_cast<MatrixWorkspace>(AnalysisDataService::Instance().retrieve(wsname.toStdString()));
-    const auto &dataX = input->x(0);
-    if (dataX.back() <= getValueOr(ranges, "peak-end-tof", 0.0) ||
-        dataX.front() >= getValueOr(ranges, "peak-start-tof", 0.0)) {
-      setPeakRange((3.0 * dataX.front() + dataX.back()) / 4.0, (dataX.front() + 3.0 * dataX.back()) / 4.0);
-      setBackgroundRange(dataX.front(), (7.0 * dataX.front() + dataX.back()) / 8.0);
-    } else {
-      setPeakRange(getValueOr(ranges, "peak-start-tof", 0.0), getValueOr(ranges, "peak-end-tof", 0.0));
-      setBackgroundRange(getValueOr(ranges, "back-start-tof", 0.0), getValueOr(ranges, "back-end-tof", 0.0));
-    }
+  disconnect(m_dblManager, SIGNAL(valueChanged(QtProperty *, double)), this, SLOT(calUpdateRS(QtProperty *, double)));
```

Comment:
why do we need to disconnect and reconnect the signals/slots? Could we just block the signals (something like this https://www.gnu.org/software/libc/manual/html_node/Blocking-for-Handler.html )

Reply:
We disconect because otherwise when we set the default properties first it tries to set the range selector in responce which fails because they are first both set to 0.0, so when we set the min first and it's greater than 0.0 it sets the min to 0.0 which is what you found, by blocking the signal we can set both independantly. I've not seen us use a blocker like that before. if there an example of it being used in mantid?

Reply:
I dont know. Will have a look later